### DISCUSSION TITLE
有关段落间距和标题间距的疑问

### DISCUSSION THREAD

**Discussion by Robotxm** [note286/xduts#9] [Q&A] [answered]
![`XAPB)~)` LK$_EU3{RD@QJ](https://user-images.githubusercontent.com/5405331/167256707-66ed4b13-3a54-414b-bdbb-774fa50ecb85.png)

```latex
图中可以看到主要是设计了两个接口类，DomainEventPublisher是消息发送服务接口。
DomainEventSender则定义了与消息中间件交互的方法，需要实现send方法将事件发布到消息中间件上。
DomainEventPublisher的具体实现需要调用DomainEventDAO来获取事件列表，并调用DomainEventSender将消息发送到消息中间件上。

至此消息发送方所有逻辑已经实现了，但是消息发送服务的调用时机还未确定。通常的做法是开启一个后台定时任务，对消息列表进行轮询，但这无疑会造成数据库资源的占用。
本项目的做法是基于Spring Boot的AOP机制，在涉及事件产生的方法上添加一个注解，并创建切面的后处理方法，在其中调用DomainEventPublisher的事件发布方法。
同时保留一个时间间隔较长的定时轮询任务，以应对那些优先级不高的事件以及兜底机制，保证事件都能被成功发送。
```

![S(@}4W~HZ39VTT39ZWMB%PA](https://user-images.githubusercontent.com/5405331/167256711-0fc4ee98-9233-44a1-807c-db7092893455.png)

```latex
即节点读取配置文件并初始化，接收传递的数据，并输出处理后的数据。节点内部的处理过程对外界来说就是黑箱，并且不同节点间可以看作是隔离的（没有共享状态）。
这意味着节点是可以方便地进行横向拓展和并行化的，这一点设计很好的契合了规则引擎需要处理大量数据的特点。当数据量变多的时候就可以创建多个节点进行并行处理。
为了实现规则节点的这些特性，本项目引入了Actor模型作为规则引擎的并发节点，将在本章\ref{Actor并发模型介绍}中做详细介绍。

在完成了对单个节点处理过程的分析之后，反观整体的规则链，其还缺少最后一个处理的过程，即消息的路由规则。一个节点处理完数据之后，需要将数据发往下一个节点。这里需要解决一个关键问题：
数据将会发往哪个节点？
```

以上两段文字是同一份文档中的不同段落，可以明显看出两部分的段间距是不一样的。后者的段间距大于段内行距。

![image](https://user-images.githubusercontent.com/5405331/167256972-42766c9b-da72-4e7f-8ade-3fd84c521da7.png)

在另一份文档中，甚至出现了段间距过大的问题。

![image](https://user-images.githubusercontent.com/5405331/167256842-22b205be-4956-4e69-9a34-4a3e34ff5594.png)

同时，标题间距的默认值看起来并不太符合学校规范。

根据学校规范，行距应为 1.5 倍。这样来看模板的格式设置可能有些小问题？

**Comment by note286** [comment]
请给出对于标题行距的具体要求，我没有在《工作手册》中找到，因此在文档里提供了接口自行修改。

**Comment by note286** [comment, marked as answer]
对于你说的段间距过大的问题，可以通过`\raggedbottom`来解决，但是对于间距并没有完美的结局办法，要么底部不对齐，间距一致，要么底部对齐，但是间距不一致。

**Comment by Robotxm** [comment]
感谢回复。

> 七、毕业设计（论文）的打印规格
论文正文页面和版面的设置规格：论文正文双面打印，为了便于装订、复制，要求每页纸的四周留有足够的空白边缘。以WORD97为例：
页面设置数据为：上3厘米、下2厘米、内侧3厘米、外侧2厘米；装订线 -- 1厘米；页眉  — 2厘米；  页脚 — 1厘米。 
版面设置数据为：文字的行间距 — 1. 5倍 ；公式的行间距 — 1. 5倍 ；字符间距 — 标准；页码数据－对称页边距。

《西安电子科技大学本科毕业设计（论文）撰写规范》中只模糊地表明“文字的行间距”为 1.5 倍，并没有分别指出标题和正文的要求。不过有预留接口就好办了。

段间距这边，尝试了 `\raggedbottom` 之后确实能达到想要的样子了。不过我之前还试过 `\setlength{\parskip}{0pt}`，似乎效果和 `\raggedbottom` 一样？

但二者从字面上来看语义应当是不一样的，不知道在解决本文提到的问题上具体有没有什么区别。

**Reply by note286** [reply]
`\parskip`控制段与段之间的垂直距离，默认值是弹性长度`0pt plus 1pt`。

`\raggedbottom`令页面在垂直方向顶部对齐，默认的`\flushbottom`令页面在垂直方向分散对齐。

> 段间距这边，尝试了 `\raggedbottom` 之后确实能达到想要的样子了。不过我之前还试过 `\setlength{\parskip}{0pt}`，似乎效果和 `\raggedbottom` 一样？

你所述的效果是由于令`\parskip`为固定长度，使得在分散对齐下无法进行弹性距离的延伸。如果是只有纯文本的页面，你所述的两种做法应该并无区别。但是页面不单单有文本，例如`\parskip`还会影响列表项之间的垂直距离。图表公式等间均有弹性长度，仅仅修改`\parskip`是不够的，图表公式等的间距也会视情况增大。如果你不喜欢分散对齐，应该选择顶部对齐，而不是设置`\parskip`。

**Reply by note286** [reply]
> 不过有预留接口就好办了。

你应该看完文档再提问啊，xduts的文档已经足够简洁了！

**Reply by note286** [reply]
在 08a53b6 中已增加设置页面垂直方向的对齐方式的接口。

**Reply by Robotxm** [reply]
感谢解惑，没注意到文档里提到了设置标题前后间距，是我疏忽了。
